src/sentry/services/hybrid_cloud/__init__.py

```diff
@@ -33,22 +38,137 @@
 
 logger = logging.getLogger(__name__)
 
+import pydantic
+
 from sentry.silo import SiloMode
 
 if TYPE_CHECKING:
     from sentry.api.base import Endpoint
 T = TypeVar("T")
 
+ArgumentDict = Mapping[str, Any]
+
 IDEMPOTENCY_KEY_LENGTH = 48
 REGION_NAME_LENGTH = 48
 
+DEFAULT_DATE = datetime.datetime(2000, 1, 1)
+
 
 class InterfaceWithLifecycle(ABC):
     @abstractmethod
     def close(self) -> None:
         pass
 
 
+def report_pydantic_type_validation_error(
+    field: pydantic.fields.ModelField, value: Any, errors: pydantic.error_wrappers.ErrorList
+) -> None:
+    with sentry_sdk.push_scope() as scope:
+        scope.set_level("warning")
+        scope.set_tag("field", field.name)
+        scope.set_extra("value_type", str(type(value)))
```

Comment:
@RyanSkonnord `set_extra()` isn't preferred anymore. `scope.set_context("pydantic", extra_payload)` is preferred: https://docs.sentry.io/platforms/python/enriching-events/context/#structured-context